sample-apps/segmentation/lib/train.py

```diff
@@ -114,3 +145,25 @@ def val_pre_transforms(self, context: Context):
 
     def val_inferer(self, context: Context):
         return SlidingWindowInferer(roi_size=(160, 160, 160), sw_batch_size=1, overlap=0.25)
+
+    def train_key_metric(self, context: Context):
+        all_metrics = dict()
+        all_metrics["train_dice"] = MeanDice(output_transform=from_engine(["pred", "label"]), include_background=False)
+        for _, (key_label, _) in enumerate(self.label_names.items()):
+            if key_label != "background":
+                all_metrics[key_label + "_dice"] = MeanDice(
+                    output_transform=from_engine(["pred_" + key_label, "label_" + key_label]), include_background=False
+                )
+        return all_metrics
+
+    def val_key_metric(self, context: Context):
+        all_metrics = dict()
+        all_metrics["val_mean_dice"] = MeanDice(
+            output_transform=from_engine(["pred", "label"]), include_background=False
```

Comment:
then you don't need these metrics.. if you fix the AsDiscreted

Reply:
This is to see the performance in Tensorboard independently for each segment